flexget/plugins/api_trakt.py

```diff
@@ -80,10 +128,16 @@ def get_access_token(account, token=None, refresh=False, re_auth=False):
                 data['refresh_token'] = acc.refresh_token
                 data['grant_type'] = 'refresh_token'
             elif token:
+                # We are only in here if a pin was specified, so it's safe to use console instead of logging
+                console('PIN authorization has been deprecated.')
                 data['code'] = token
                 data['grant_type'] = 'authorization_code'
             else:
-                raise plugin.PluginError('Account %s not found in db and no pin specified.' % account)
+                log.debug('No pin specified for an unknown account %s. Attempting to authorize device.' % account)
+                try:
+                    device_auth(account, session)
```

Comment:
Shouldn't this return after, instead of continuing to the token submit? Also, there isn't logic inside there when adding a new auth to check if there is an existing db entry already for that account. Maybe we should have 2 helper methods that return the access and refresh tokens for pin and device auth, then keep the db storage in the same method.